web/packages/shared/components/MenuLoginWithActionMenu/MenuLoginWithActionMenu.tsx

```diff
@@ -0,0 +1,138 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ReactElement, useRef, useState } from 'react';
+
+import { ButtonBorder, Flex, Menu, MenuItem } from 'design';
+import { ButtonSize } from 'design/Button';
+import * as icons from 'design/Icon';
+import {
+  LoginItem,
+  MenuInputType,
+  MenuLogin,
+} from 'shared/components/MenuLogin';
+
+/**
+ * Displays a menu button with additional action menu attached to the right of it.
+ * <MenuLoginWithActionMenu> expects two children, a LoginItem for <MenuLogin> and <MenuItem>
+ * for action memnu.
+ *
+ * @param buttonText - button text for main menu button.
+ * @param onSelect - handles select or click in main menu items.
+ * @param getLoginItems - fetches login items.
+ * @param children - action menu items.
+ * @param placeholder - text for action menu search box or static label.
+ * @param size - button icon size.
+ * @param width - button width.
+ * @param disableSearchAndFilter - instructs underlying MenuLogin component to display static
```

Comment:
I don't think these JSDoc param tags are correct, as AFAIR `@param` when used like this is for positional arguments and `MenuLoginWithActionMenu` has only one positional argument.

Even if they were updated to the correct form (see [Parameters with properties](https://jsdoc.app/tags-param#parameters-with-properties)), I'm not sure if most editor integrations are able to correctly parse them. Usually I use an individual JSDoc comment for each prop instead:

https://github.com/gravitational/teleport/blob/3d22adc854c3038da5bfd66e71dee2af66b2d20e/web/packages/teleterm/src/ui/components/OfflineGateway.tsx#L29-L51

Reply:
Agree https://github.com/gravitational/teleport/pull/52877/commits/59bf9a2547973be133789597214d6ddc08ac496c